video/out/vo_gpu_next.c

```diff
@@ -714,7 +718,36 @@ static bool map_frame(pl_gpu gpu, pl_tex *tex, const struct pl_source_frame *src
         frame->color.hdr.max_luma =
             pl_hdr_rescale(PL_HDR_PQ, PL_HDR_NITS, color->source_max_pq / 4095.0f);
     }
-#endif
+
+#if PL_API_VER >= 246 && HAVE_LIBDOVI
+    if (mpi->dovi_buf) {
+        DoviRpuOpaque *rpu_in =
+            dovi_parse_unspec62_nalu(mpi->dovi_buf->data, mpi->dovi_buf->size);
+        const DoviRpuDataHeader *header = dovi_rpu_get_header(rpu_in);
+
+        if (header && header->vdr_dm_metadata_present_flag) {
+            const DoviVdrDmData *vdr_dm_data = dovi_rpu_get_vdr_dm_data(rpu_in);
+            if (vdr_dm_data->dm_data.level1) {
+                const DoviExtMetadataBlockLevel1 *l1 = vdr_dm_data->dm_data.level1;
+                const float max_luma =
+                    pl_hdr_rescale(PL_HDR_PQ, PL_HDR_NITS, l1->max_pq / 4095.0f);
+
+                for (int i = 0; i < 3; i++)
+                    frame->color.hdr.scene_max[i] = max_luma;
+
+                frame->color.hdr.scene_avg =
+                    pl_hdr_rescale(PL_HDR_PQ, PL_HDR_NITS, l1->avg_pq / 4095.0f);
+            }
+
+        }
+
+        if (header)
+            dovi_rpu_free_header(header);
+        dovi_rpu_free(rpu_in);
```

Comment:
Why this asymmetry? You check for null in libdovi, so you could skip check here. Or... check both. Functionally no difference, just looks odd.